src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todos } from './types';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const preparedTodos: Todos[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(({ id }) => id === todo.userId) || null,
+}
+));
 
 export const App = () => {
+  const [titleTodo, setTitleTodo] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [todos, setTodos] = useState<Todos[]>(preparedTodos);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const resetForm = () => {
+    setTitleTodo('');
+    setUserId(0);
+  };
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setTitleTodo(event.target.value);
+
+    if (titleError) {
+      setTitleError(false);
+    }
+  }
+
+  function handleUserSelect(event: React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+event.target.value);
+
+    if (userIdError) {
+      setUserIdError(false);
+    }
+  }
+
+  function handleButtonAdd(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    let isError = false;
+
+    if (!titleTodo) {
+      setTitleError(true);
+      isError = true;
+    }
+
+    if (!userId) {
+      setUserIdError(true);
+      isError = true;
+    }
+
+    if (isError) {
+      return;
+    }
```

Comment:
Will this solution work?
```suggestion
    if (!titleTodo || !userId) {
      setTitleError(!titleTodo);
      setUserIdError(!userId);
      return;
    }
```